platform/ios/src/MGLAnnotationView.m

```diff
@@ -56,6 +62,18 @@ - (void)updatePitch:(CGFloat)pitch
     self.layer.transform = t;
 }
 
+- (void)updateScaleForPitch:(CGFloat)pitch maxPitch:(CGFloat)maxPitch
+{
+    CGFloat superViewHeight = CGRectGetHeight(self.superview.frame);
+    if (superViewHeight > 0.0) {
+        CGFloat scaleReduction = 1.0 - self.center.y / superViewHeight;
+        CGFloat pitchIntensity = MAX(pitch, 0) / maxPitch;
+        CGFloat pitchAdjustedScale = 1.0 - scaleReduction * pitchIntensity;
```

Comment:
Can you add comments explaining the rationale behind this formula, or alternatively where this formula came from?